File: resources/android/{{ plugin }}Functions.kt
package com.{{ vendor }}.{{ package }}

import com.nativephp.mobile.bridge.BridgeFunction
import org.json.JSONObject

/**
 * Android bridge functions for {{ vendor }}/{{ package }}.
 *
 * Flow:
 * PHP calls {{ plugin }}::example()
 * NativePHP invokes {{ plugin }}.Example
 * Kotlin receives the JSON payload here
 * Android APIs can be called from execute()
 * A JSONObject is returned to PHP
 */
class {{ plugin }}Functions {
    class Example : BridgeFunction() {
        override fun execute(payload: JSONObject): JSONObject {
            val response = JSONObject()

            response.put("plugin", "{{ plugin }}")
            response.put("platform", "android")
            response.put("received", payload)

            return response
        }
    }
}

File: resources/ios/{{ plugin }}Functions.swift
import Foundation
import NativePHP

/// iOS bridge functions for {{ vendor }}/{{ package }}.
///
/// Flow:
/// PHP calls {{ plugin }}::example()
/// NativePHP invokes {{ plugin }}.Example
/// Swift receives the payload here
/// iOS APIs can be called from execute()
/// A dictionary is returned to PHP
public enum {{ plugin }}Functions {
    public final class Example: BridgeFunction {
        public init() {}

        public func execute(payload: [String: Any]) async throws -> Any {
            return [
                "plugin": "{{ plugin }}",
                "platform": "ios",
                "received": payload,
            ]
        }
    }
}
